autokeras/hypermodel/preprocessor.py

```diff
@@ -673,3 +676,233 @@ def set_config(self, config):
+
+        # append num-cat value
+        for key, value in self.high_level_num_cat.items():
+            num_col_index, cat_col_index = key
+            cat_name = str(x[cat_col_index])
+            new_value = value[cat_name] if cat_name in value else -1
+            new_values.append(new_value)
+
+        # LabelEncoding
+        for col_index in self.categorical_col:
+            key = str(x[col_index])
+            try:
+                x[col_index] = self.label_encoders[col_index].transform(key)
+            except KeyError:
+                x[col_index] = -1
+        return np.hstack((x, np.array(new_values)))
+
+    def _impute(self, x):
+        for col_index in range(self.num_columns):
+            if col_index in self.numerical_col:
+                if x[col_index] == 'nan':
+                    x[col_index] = 0.0
+                else:
+                    x[col_index] = float(x[col_index])
+            else:
+                if x[col_index] == 'nan':
+                    x[col_index] = 0
+
+    def finalize(self):
+        # divide column according to category number of the column
+        for col_index in self.value_counters.keys():
+            num_cat = len(self.value_counters[col_index])
+            # if num_cat <= 32:
+            #     continue
+            # calculate frequency
+            if num_cat > 32 and num_cat <= 100:
+                self.high_level1_col.append(col_index)
+                self.count_frequency[col_index] = {
+                    key: value / (self.num_rows * num_cat)
+                    for key, value in self.value_counters[col_index].items()}
+            elif num_cat > 100:
+                self.high_level2_col.append(col_index)
+
+        self.high_level2_col.sort()
+
+        for index, cat_col_index1 in enumerate(self.high_level2_col):
+            # extract high level columns from cat-cat dict
+            for cat_col_index2 in self.high_level2_col[index + 1:]:
+                pair = (cat_col_index1, cat_col_index2)
+                self.high_level_cat_cat[pair] = self.categorical_categorical[pair]
+
+            # extract high level columns from num-cat dict and calculte mean
+            for num_col_index in self.numerical_col:
+                pair = (num_col_index, cat_col_index1)
+                self.high_level_num_cat[pair] = self.numerical_categorical[pair]
+                for key, value in self.high_level_num_cat[pair].items():
+                    self.high_level_num_cat[pair][key] /= self.value_counters[
+                        cat_col_index1][key]
+
+        self._shape = (len(self.column_types)
+                       + len(self.high_level1_col)
+                       + len(self.high_level_cat_cat)
+                       + len(self.high_level_num_cat),)
+
+    def output_types(self):
+        return (tf.float64,)
+
+    @property
+    def output_shape(self):
+        return self._shape
+
+    def get_weights(self):
+        return {'_shape': self._shape,
```

Comment:
Config dicts should be considered as part of the public API. As such they should not contain private symbol. Consider removing the underscore here.